python/lsst/rubintv/production/pipelineRunning.py

```diff
@@ -206,12 +217,12 @@ def callback(self, payload: Payload) -> None:
         self.log.debug(f"{self.step=} {self.dataProduct=}")
 
         try:
-            if pipelineGraphBytes is not None and pipelineGraphBytes != self.pipelineGraphBytes:
-                self.log.warning("Pipeline graph has changed, updating")
-                self.pipelineGraphBytes = pipelineGraphBytes
-                self.pipelineGraph = pipelineGraphFromBytes(pipelineGraphBytes)
-                # need to remake the caching butler if the pipeline changes
-                self.cachingButler = self.makeCachingLimitedButler(self.butler)
+            # NOTE: if someone sends a pipelineGraphBytes that's so different
+            # from the pipelines built by buildPipelines that it consumes
+            # different data products, things won't be cached, but assuming
+            # that's not the case this should be OK. Otherwise, remake the
+            # CachingLimitedButler with the new pipelineGraph here.
+            pipelineGraph = pipelineGraphFromBytes(pipelineGraphBytes)
```

Comment:
@TallJimbo this is the bit I need checked to make sure I'm not misunderstanding.

Reply:
It's not just the input data products that won't be cached but any outputs that are also different (those are cached on write).  But you're right that the worst that will happen is that no caching will occur.  I don't see any evidence that the code will "remake the CachingLimited with the new pipelineGraph" as the comment says, though.